### PR TITLE
refactor(core): eliminate business profile database queries in payments confirm flow

### PR DESCRIPTION
**Pull request by BallaNitesh**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Removing db calls which are unnecessary

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/core/utils.rs [hunk 1]
```diff
@@ -992,16 +997,19 @@ pub async fn get_profile_id_from_business_details(
     merchant_account: &domain::MerchantAccount,
     request_profile_id: Option<&String>,
     db: &dyn StorageInterface,
+    should_validate: bool,
```
Comment (Narayanbhat166) [inline crates/router/src/core/utils.rs:1000]: validate only in case of payments and mca create, in other cases validation is not needed

**Review by Narayanbhat166 — CHANGES_REQUESTED** [review]

File: crates/router/src/core/utils.rs (hunk 1 above)
Comment (BallaNitesh) [reply crates/router/src/core/utils.rs:1000]: Is it required in list payment methods also?

Comment (Narayanbhat166) [reply crates/router/src/core/utils.rs:1000]: No its not required in payment methods list. The entry point is payments create and once a profile id is given for a payment, it cannot be changed in any other flow

**Review by inventvenkat — CHANGES_REQUESTED** [review]
What is this should validate is been passed everywhere?

**Review by inventvenkat — APPROVED** [review]

**Pull request merged**